File: module_4_practice/DiscountCalculator.java
public class DiscountCalculator {
    private DiscountStrategy discountStrategy;

    public DiscountCalculator(DiscountStrategy discountStrategy) {
        this.discountStrategy = discountStrategy;
    }

    public double calculateDiscount(double price) {
        return discountStrategy.applyDiscount(price);
    }
}

File: module_4_lab/OCP/DiscountCalculator.cs
public class DiscountCalculator
{
    private readonly IDiscountStrategy _discountStrategy;

    public DiscountCalculator(IDiscountStrategy discountStrategy)
    {
        _discountStrategy = discountStrategy;
    }

    public double Calculate(double amount)
    {
        return _discountStrategy.CalculateDiscount(amount);
    }
}